### PR TITLE
Fix issue with SQL auditing policy casing

### PR DESCRIPTION
**Pull request by jaredfholgate**
## Overview/Summary

This resolves a casing issue with the `logAnalyticsWorkspaceId` parameter for the `Deploy-AzSqlDb-Auditing` policy assignment.   Although the different case is being used in many places in the module, this fix has been scoped to just this particular assignment to avoid any impact outside of this bug report.

## This PR fixes/adds/changes/removes

1. #756 

### Breaking Changes

None

## Testing Evidence

Pre-fix if the policy was updated in the UI:

```
  # module.core.module.alz.azurerm_management_group_policy_assignment.enterprise_scale["/providers/Microsoft.Management/managementGroups/myorg-landing-zones/providers/Microsoft.Authorization/policyAssignments/Deploy-AzSqlDb-Auditing"] will be updated in-place
  ~ resource "azurerm_management_group_policy_assignment" "enterprise_scale" {
        id                   = "/providers/Microsoft.Management/managementGroups/myorg-landing-zones/providers/Microsoft.Authorization/policyAssignments/Deploy-AzSqlDb-Auditing"
        name                 = "Deploy-AzSqlDb-Auditing"
      ~ parameters           = jsonencode(
          ~ {
              ~ logAnalyticsWorkspaceId = {
                  ~ value = "/subscriptions/11eb2a49-f2be-48ee-a1b3-113a01d79ffc/resourcegroups/myorg-mgmt/providers/microsoft.operationalinsights/workspaces/myorg-la" -> "/subscriptions/11eb2a49-f2be-48ee-a1b3-113a01d79ffc/resourceGroups/myorg-mgmt/providers/Microsoft.OperationalInsights/workspaces/myorg-la"
```

Post-fix the plan does not contain the update as desired/

Here is what it looks like now:
image

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.

### CODE REVIEW

File: modules/management/locals.tf [hunk 1]
```diff
@@ -233,7 +233,7 @@ locals {
     "${local.root_id}-landing-zones" = {
       parameters = {
         Deploy-AzSqlDb-Auditing = {
-          logAnalyticsWorkspaceId = local.log_analytics_workspace_resource_id
+          logAnalyticsWorkspaceId = lower(local.log_analytics_workspace_resource_id)
```
Comment (jaredfholgate) [inline modules/management/locals.tf:236]: @matt-FFFFFF It turns out this local just overrides the settings from the policy assignment archetype anyway, so may as well just have been an empty string in the other two files updated here...

File: .github/scripts/Invoke-LibraryUpdatePolicyAssignments.ps1 [hunk 2]
```diff
@@ -61,7 +61,7 @@ $defaultParameterValues =@(
     "-p automationRegion=`${default_location}",
     "-p retentionInDays=30",
     "-p rgName=`${root_scope_id}-mgmt",
-    "-p logAnalyticsResourceId=/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/`${root_scope_id}-mgmt/providers/Microsoft.OperationalInsights/workspaces/`${root_scope_id}-la",
+    "-p logAnalyticsResourceId=/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/`${root_scope_id}-mgmt/providers/microsoft.operationalinsights/workspaces/`${root_scope_id}-la",
```
Comment (matt-FFFFFF) [inline .github/scripts/Invoke-LibraryUpdatePolicyAssignments.ps1:64]: Hi @jaredfholgate 

If we change the case for all assignments, as appears to be the case here - would all assignments that use this value be re-created?

File: modules/management/locals.tf (hunk 1 above)
Comment (matt-FFFFFF) [reply modules/management/locals.tf:236]: Could we just have this in as the workaround? Leaving PascalCase for everything else?

**Review by matt-FFFFFF — CHANGES_REQUESTED** [review]

File: .github/scripts/Invoke-LibraryUpdatePolicyAssignments.ps1 (hunk 2 above)
Comment (jaredfholgate) [reply .github/scripts/Invoke-LibraryUpdatePolicyAssignments.ps1:64]: This parameter is literally only used by that one policy, so there is no impact outside of that one. That one policy is an outlier, all others construct the id based on parameters, whereas this one takes the whole id as a parameter.

File: modules/management/locals.tf (hunk 1 above)
Comment (jaredfholgate) [reply modules/management/locals.tf:236]: As with the other comment, this is literally just used by that one policy. All others construct the id based on a workspace name passed in.

**Comment by jaredfholgate** [comment]
@matt-FFFFFF See here: https://github.com/search?q=repo%3AAzure%2FEnterprise-Scale%20logAnalyticsWorkspaceId%20&type=code

That is the only policy that accepts the workspace id in this format. I think this probably needs resolving upstream eventually, but this workaround will do the trick for now.

**Comment by matt-FFFFFF** [comment]
Thanks @jaredfholgate LGTM

**Review by matt-FFFFFF — APPROVED** [review]

**Comment by matt-FFFFFF** [comment]
AB#29528

**Pull request merged**